documentation/library_guide/kernel_templates/esimd/radix_sort.rst

```diff
@@ -0,0 +1,285 @@
+
+         sycl::host_accessor v_acc{values, sycl::read_only};
+         for(std::size_t i = 0; i < n; ++i)
+               std::cout << v_acc[i] << ' ';
+         std::cout << '\n';
+      }
+
+      return 0;
+   }
+
+**Output:**
+
+.. code:: none
+
+   1 2 3 3 3 5
+   s o r t e d
+
+
+Recommended settings for the best performance
+---------------------------------------------
+
+General advice is to set the configuration according to the performance measurements and profiling information.
+
+But the initial configuration may be selected according to these points:
+
+a. The number of elements to sort is small (~16K or less) and the algorithm is ``radix_sort``. The elements can be processed by a single work-group.
+
+   - Increase ``param`` values, so ``N <= param.data_per_workitem * param.workgroup_size``.
+
+b. The number of elements to sort is medium (between ~16K and ~1M). All the work-groups can execute simultaneously.
+
+   - Make sure the device is saturated: ``param.data_per_workitem * param.workgroup_size ≈ N / device_xe_core_count``. Prefer larger ``param.workgroup_size`` in ``param.data_per_workitem * param.workgroup_size`` combination to reduce the number of work-groups and thus synchronization overhead.
+
+c. The number of elements to sort is large (more than ~1M). The work-groups preempt each other.
+
+   - Increase the occupancy to hide the latency: ``param.data_per_workitem * param.workgroup_size ≈< N / (device_xe_core_count * desired_occupancy)``. The occupancy depends on the local memory usage which is determined by ``key_type``, ``value_type``, ``radix_bits``, ``param.data_per_workitem`` and ``param.workgroup_size`` parameters. Refer to "Local memory usage" chapter for the calculation.
+
+
+Limitations (may be relaxed in the future)
+------------------------------------------
+
+- Algorithms can process only C++ integral and floating-point types with the width up to 64-bits (except for ``bool``).
+- Number of elements to sort must not exceed `2^30`.
+- ``radix_bits`` can only be `8`.
+- ``param.data_per_workitem`` has discreteness of `32`.
+- ``param.workgroup_size`` can only be `64`.
+- Local memory is always used to rank keys, reorder keys or key-value pairs which limits possible values of ``param.data_per_workitem`` and ``param.workgroup_size``.
+- ``radix_sort_by_key`` does not have single-work-group implementation yet.
+
+
+Possible API extensions (may be implemented in the future)
+----------------------------------------------------------
+
+- Allow passing externally allocated memory.
+- Allow passing dependent events.
+- Allow passing a range of bits to sort.
+- Allow out-of-place sorting, e.g. with a double-buffer or an output sequence(s)
+- Allow configuration of kernels other than the most time-consuming kernel (e.g. of a kernel computing histograms).
+- Allow range transformations (e.g. range pipes or transform iterators).
+
+
+System requirements (coverage my be extended in the future)
```

Comment:
Change to System Requirements. If it is important for user understanding, please add a sentence stating that the coverage may be extended in the future.

Reply:
Done.